### DISCUSSION TITLE
Configuration of max post size?

### DISCUSSION THREAD

**Discussion by scottharman** [tobychui/zoraxy#387] [Q&A]
Hey team - I've got a weird issue that running a git server behind zoraxy - how do we set the max body size to prevent timeouts?
Recommended nginx proxy settings are as below - presumably I can just use custom headers for some of these, but max body size is the main issue.
I see that I can set the X- values via custom headers, but is the max body size available, particularly for LFS.

        client_max_body_size 512M;
        proxy_set_header Connection $http_connection;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

**Comment by tobychui** [comment]
Hi @scottharman 

There are no team, for the proxy function in zoraxy, it just me doing all the work and you can call me Toby :D 

>  how do we set the max body size to prevent timeouts?

For your information, there are no max body size limit in Zoraxy. Zoraxy use a stream based io copy method for proxying (i.e. for every byte it see from upstream, it forward it to downstream). So your timeout are mostly caused by the transport round tripper. 

In the current implementation, Zoraxy is using the default Golang net/http HTTP transport system for the request handling. These settings are per-host based (aka each proxy rule have one of these transport handler). The default value shd looks like this.

```go
// DefaultTransport is the default implementation of [Transport] and is
// used by [DefaultClient]. It establishes network connections as needed
// and caches them for reuse by subsequent calls. It uses HTTP proxies
// as directed by the environment variables HTTP_PROXY, HTTPS_PROXY
// and NO_PROXY (or the lowercase versions thereof).
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: defaultTransportDialContext(&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}),
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}

```

I think I can add a customizable timeout duration for each host, however as this is inside the `dpcore` module (instead of the interfacing module `dynamicProxy` that provides API to outer layers of the code base), this change might need to take some time and discussion. If you are interested in this function, please open a formal enhancement request in the issue page.
